### PR TITLE
Rebase consistency

### PR DESCRIPTION
**Pull request by myronkscott**
need more tests.  for review only.

### CODE REVIEW

File: voter/src/main/java/org/terracotta/voter/ClusterState.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import com.terracotta.connection.api.DiagnosticConnectionService;
+import com.terracotta.diagnostic.Diagnostics;
+import java.net.URI;
+import java.util.Properties;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ */
+public class ClusterState {
```
Comment (albinsuresh) [inline voter/src/main/java/org/terracotta/voter/ClusterState.java]: To be removed in the final PR.

File: voter/src/main/java/org/terracotta/voter/VetoVote.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ */
+package org.terracotta.voter;
+
+import java.util.Arrays;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ */
+public class VetoVote {
```
Comment (albinsuresh) [inline voter/src/main/java/org/terracotta/voter/VetoVote.java]: To be removed in the final PR.

File: dso-l2/src/main/java/com/tc/l2/state/ConsistencyManagerImpl.java [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    }
+    return allow;
+  }
+  
+  private int voteThreshold(ServerMode mode) {
+    //  peer servers plus extra votes plus self is the total votes available
+    int voteCount = peerServers + voter.getVoterLimit() + 1;
+    if (mode == ServerMode.ACTIVE) {
+      // only half the votes needed to continue as active
+      return voteCount - (voteCount>>1) - 1; // ceiling of half minus the vote for self
```
[the remark this answers is not in the document]
Comment (myronkscott) [reply dso-l2/src/main/java/com/tc/l2/state/ConsistencyManagerImpl.java]: The threshold is correct, -1 is for the server you are on which always votes for itself.  You need 2 out of 4 outside votes to win the tally.

File: dso-l2/src/main/java/com/tc/l2/state/StateManagerImpl.java [hunk 4]
```diff
@@ -86,6 +88,23 @@ public StateManagerImpl(Logger consoleLogger, GroupManager<AbstractGroupMessage>
...
+    map.put("startState", this.startState);
+    map.put("currentState", this.state);
+    map.put("active", this.activeNode);
+    map.put("syncedTo", this.syncdTo);
+    if (this.availabilityMgr instanceof ConsistencyManagerImpl) {
+      ConsistencyManagerImpl cc = (ConsistencyManagerImpl)this.availabilityMgr;
+      map.put("requestedActions", cc.getActions());
+      map.put("availabilityRestriction", cc.isVoting());
+    } else {
+
```
[the remark this answers is not in the document]
Comment (myronkscott) [reply dso-l2/src/main/java/com/tc/l2/state/StateManagerImpl.java]: yes.  no information if it is not the implementation

File: dso-l2/src/main/java/com/tc/net/groups/TCGroupManagerImpl.java [hunk 5]
```diff
@@ -820,23 +820,27 @@ private boolean isZappedNode(NodeID nodeID) {
...
+      memberReport.put(entry.getKey().toString(), entry.getValue());
     }
 
-    List<String> zapped = new ArrayList<>(this.zappedSet.size());
+    List<Object> zapped = new ArrayList<>(this.zappedSet.size());
     map.put("zapped", zapped);
-    this.zappedSet.forEach(node->zapped.add(node.toString()));
+    this.zappedSet.forEach(node->zapped.add(node));
+    } catch (RuntimeException r) {
+      r.printStackTrace();
```
[the remark this answers is not in the document]
Comment (myronkscott) [reply dso-l2/src/main/java/com/tc/net/groups/TCGroupManagerImpl.java]: will change to a log statement

File: voter/src/main/java/org/terracotta/voter/ActiveVoter.java [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+    try {
+      return registrationLatch.get();
+    } catch (ExecutionException e) {
+      throw new RuntimeException(e);
+    } finally {
+      futures.forEach(f->f.cancel(true));
+    }
+  }
+
+  public void registerAndHeartbeat(String id, ExecutorService executorService, ClientVoterManager currentTerm, List<ClientVoterManager> voterManagers) throws InterruptedException {
```
[the remark this answers is not in the document]
Comment (myronkscott) [reply voter/src/main/java/org/terracotta/voter/ActiveVoter.java]: will change

File: voter/src/main/java/org/terracotta/voter/ClusterState.java (hunk 1 above)
Comment (myronkscott) [reply voter/src/main/java/org/terracotta/voter/ClusterState.java]: Are you opposed?  I thought it useful as an internal tool

File: voter/src/main/java/org/terracotta/voter/VetoVote.java (hunk 2 above)
Comment (myronkscott) [reply voter/src/main/java/org/terracotta/voter/VetoVote.java]: Are you opposed?  Useful to me as an internal tool.  Would like is accessible for integration tests.

**Comment by cschanck** [comment]
OK, that's nice and chunky.

File: voter/src/main/java/org/terracotta/voter/VetoVote.java (hunk 2 above)
Comment (albinsuresh) [reply voter/src/main/java/org/terracotta/voter/VetoVote.java]: Oh I thought you just forgot to remove it after your testing. But if it was intentional then I'm fine with keeping it internally.

BTW, If it's intended to be used in integration tests as well then just wondering if the direct logback logger usage would clash with the logger impl used in integration tests. If they're the same the this should be fine.

File: voter/src/main/java/org/terracotta/voter/ClusterState.java (hunk 1 above)
Comment (albinsuresh) [reply voter/src/main/java/org/terracotta/voter/ClusterState.java]: Fine with keeping it for internal use.

**Pull request merged**